src/app/store/accounts/submitted-transactions.hooks.ts

```diff
@@ -0,0 +1,58 @@
+import { useAtom } from 'jotai';
+import { useCallback } from 'react';
+
+import { deserializeTransaction } from '@stacks/transactions';
+
+import { accountSubmittedTransactionsState, SubmittedTransaction } from './submitted-transactions';
+import { useCurrentAccountTxIds } from '@app/query/transactions/transaction.hooks';
+import { safelyFormatHexTxid } from '@app/common/utils/safe-handle-txid';
+
+export function useAccountSubmittedTransactionsState() {
+  return useAtom(accountSubmittedTransactionsState);
+}
+
+export function useAddSubmittedTransactionCallback() {
+  const [submittedTxs, setSubmittedTxs] = useAccountSubmittedTransactionsState();
+  return useCallback(
+    ({ rawTx, txid }: SubmittedTransaction) => {
+      setSubmittedTxs([{ rawTx, txid }, ...submittedTxs]);
+    },
+    [setSubmittedTxs, submittedTxs]
+  );
+}
+
+export function useRemoveSubmittedTransactionCallback() {
+  const [submittedTxs, setSubmittedTxs] = useAccountSubmittedTransactionsState();
+  return useCallback(
+    (txid: string) => {
+      const updatedSubmittedTxs = submittedTxs.filter(tx => tx.txid !== txid);
+      setSubmittedTxs([...updatedSubmittedTxs]);
+    },
+    [setSubmittedTxs, submittedTxs]
+  );
+}
+
+export function useAccountSubmittedStacksTransactions() {
+  const [submittedTxs] = useAccountSubmittedTransactionsState();
+  return submittedTxs.map(tx => deserializeTransaction(tx.rawTx));
+}
```

Comment:
we should memoize these hooks. hooks returning new obj reference can be hard to track down when they cause render loops. also imagine deserializetx to be a fairly expensive operation that's happening on every render where this is used